src/Storages/MergeTree/ReplicatedMergeTreePartCheckThread.cpp

```diff
@@ -179,10 +179,15 @@ ReplicatedMergeTreePartCheckThread::MissingPartSearchResult ReplicatedMergeTreeP
                     found_part_with_the_same_min_block = true;
                     parts_found.push_back(part_on_replica);
                 }
+
                 if (part_on_replica_info.max_block == part_info.max_block)
                 {
                     found_part_with_the_same_max_block = true;
-                    parts_found.push_back(part_on_replica);
+
+                    /// If we are looking for part like partition_X_X_level we can add part
+                    /// partition_X_X_(level-1) two times, avoiding it
+                    if (parts_found.empty() || parts_found.back() != part_on_replica)
```

Comment:
Looks stupid:
```
Found parts with the same min block and with the same max block as the missing part 1_332_332_1 on replica r_0. Hoping that it will eventually appear as a result of a merge. Parts: 1_332_332_0, 1_332_332_0
```